### ISSUE TITLE
Set packagecomponent flag on Products

### ISSUE THREAD

**Issue by i-rabot** [kbower/tickettest1#3289]
_Imported from trac ticket 3289.
Created by **mariam**
Opened in trac: Jul 6, 2015, 4:44:41 PM
Last modified in trac: Oct 26, 2016, 4:25:37 PM
Freshdesk: [1880](https://retailarchitects.freshdesk.com/helpdesk/tickets/1880)_
***
_by mariam:_
Packages created in Loft are not setting the packagecomponent to Y when the component is added to a package.  Without the flag set, users cannot view the packages a component is a part of in GERs.  
**Need to set the flag in Product to Y for a product when it is added to packages as a component.**

**Comment by i-rabot** [comment]
_**kbower** trac comment on Jul 7, 2015, 8:22:54 AM:_

Why GERS, why?  It is unfortunate that GERS is so poor at database design.  The "flag" of whether a product is a component is whether it is part of a package, not a Y/N flag that needs to be kept in sync with whether it is part of a package. (!)

So, to implement this means not only must it be set, but it must be properly unset when it is removed from (all) packages.  Removing the product from a package must check to see if it was part of any other package and, if not, set component flag to N.

**Comment by i-rabot** [comment]
_**kbower** trac comment on Jul 7, 2015, 8:24:04 AM:_

UNLESS, the flag is meant as a "can act as component" flag, in which case it should stay Y.  We need to find out what the flag means: can you manually set it to Y, or does only GERS set it?  If you can manually set it to Y, that changes the design.

**Comment by i-rabot** [comment]
_**kbower** trac comment on Jul 7, 2015, 8:26:41 AM:_

Need answer on above.

**Comment by i-rabot** [comment]
_**kbower** trac status change on Jul 7, 2015, 8:26:41 AM:_ **waiting**

**Comment by i-rabot** [comment]
_**sjohnson** trac comment on Jul 7, 2015, 9:36:50 AM:_

Mariam,
Before we can "fix" this, we need to know what it really means.  Do you set that manually in GERS meaning it can act as a component or does it automatically get set behind the scenes when you add it to a package?  What happens if you remove it from all packages, does it get reset back to N?
Thanks,
Scott
___
Scott,
We do not set this manually, this happens automatically. If the component is removed from all existing packages, it gets reset back to N.
Mariam

**Comment by i-rabot** [comment]
_**sjohnson** trac comment on May 9, 2016, 9:56:26 AM:_

Ticket [#3577](3577) has been marked as a duplicate of this ticket.

**Comment by i-rabot** [comment]
_**kbower** trac comment on Oct 26, 2016, 2:46:53 PM:_

Must be done as after insert & after delete sqlalchemy events on Package class.

**Comment by i-rabot** [comment]
_**kbower** trac comment on Oct 26, 2016, 4:12:26 PM:_

knock this out quick...

**Comment by i-rabot** [comment]
_**kbower** trac comment on Oct 26, 2016, 4:25:37 PM:_

(Changeset [[7813]](../commit/2bf29fd1096868908f44db2030d988da72956415))
* fixes #3289, Set packagecomponent flag on Products based on whether the package is being used as a component #all
   * also update all existing products that are set incorrectly during release upgrade

**Comment by i-rabot** [comment]
_**kbower** trac status change on Oct 26, 2016, 4:25:37 PM:_ **closed**
